docs/metrics/llm-prompt-based-metric.md

```diff
@@ -0,0 +1,174 @@
+# LLM Prompt-based Metric
+
+Prompting Engineering is a technique that involves building and optimizing specific input prompts to allow our Large
+Language Model (LLM) to perform better at targeted tasks. In the landscape of LLM evaluations, prompt engineering is
+designed to envoke the LLMs inherently knowledge and ability to reason to determine whether or not a set of inputs
+contain some sort of hallucination.
+
+## LLM Configurations
+
+Before diving into the techniques used for prompting we first need to understand some of the common terminology used as
+parameters for LLMs.
+
+**Temperature** - Determines how the randomness of the model. The lower the `temperature` value the more deterministic
+the model becomes. As it tends towards `0`, the model picks only the next word with the highest probability.
+
+**Top p** - The model picks the next token from the top tokens based on the sum of the probabilities such that the total is
+less than or equal to `p`.
+
+##  Implementation Details
+
+We decide to use [GPT-4](https://openai.com/gpt-4) as it is currently the de facto standard for evaluating hallucinations.
+Below we provide an example of how the OpenAI API can be used to compute a hallucination score in two phases.
+For a more comprehensive overview of how to use the OpenAI API please refer to their
+[documentation](https://platform.openai.com/docs/overview).
+
+!!! info "LLM Prompting for RAG Type"
+
+    The presented example outlines LLM prompting for Non-RAG Type problems like TruthfulQA. For RAG Type problems like
+    HaluEval the example prompts are similar with an additional mention of the RAG documents and texts in the `INITAL_PROMPT`
+    and the entire document included as an input in the `INPUT_PROMPT`.
+
+
+### Phase 1 ~ Boolean Format Prompt
```

Comment:
Given that the boolean prompt works better than the score prompt, I think we should only keep the boolean format prompt and add a callout saying that a score can be prompted instead but boolean gives more accurate results and the score it gives out is not exactly explainable.